### PR TITLE
Add endpoint for creating a policy acceptance for a specific user

### PR DESCRIPTION
**Pull request by dati18**
**Summary**
- Add authenticated endpoint to record policy acceptances for the current user, with strict validation and all-or-nothing behavior for invalid policy IDs.

**Behavior**
- Uses authenticated user from session context (no `user_id` in payload)
- Ignores already-accepted policies (idempotent writes via `firstOrCreate`)
- If any provided policy ID does not exist, returns `400` with `missing_policy_ids`
- No acceptances are created when invalid policy IDs are present
- Returns `422` for invalid payload (missing `policy_ids`, non-array, non-integer items)

Bug: T431674

### CODE REVIEW

File: tests/Routes/PolicyAcceptanceControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    public function testUnauthenticatedRequestResponds401(): void {
+        $this->json('PUT', $this->route)
+            ->assertStatus(401);
+    }
+
+    public function testAcceptSinglePolicy(): void {
+        $user = User::factory()->create();
+        $policy = $this->makePolicy();
+
+        $this->actingAs($user, 'api')
```
Comment (tarrow) [inline tests/Routes/PolicyAcceptanceControllerTest.php:37]: I had to check what `api` means in this context. Turns out it is the name of the `guard`. Is it necessary to specify the guard here?

I do see we do this in most other places (exception is LimitWikiAccessTest) so probably we should just keep this here. I don't think you should change anything but I'd be interested if you know the answer to the question.

File: tests/Routes/PolicyAcceptanceControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            ->assertJsonFragment(['missing_policy_ids' => [$nonExistentId]]);
+
+        // Nothing should have been written
+        $this->assertDatabaseMissing('policy_acceptances', [
+            'user_id' => $user->id,
+            'policy_id' => $policy->id,
+        ]);
+    }
+
+    public function testMissingPolicyIdsFieldReturns422(): void {
```
Comment (tarrow) [inline tests/Routes/PolicyAcceptanceControllerTest.php:125]: We didn't specify it in the ticket but I am happy that you kept the same behaviour as other endpoints for validation failures.

File: app/Http/Controllers/PolicyAcceptanceController.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        ]);
+
+        $policyIds = $request->input('policy_ids');
+
+        // Check all policy IDs exist before writing anything
+        $existingIds = Policy::whereIn('id', $policyIds)->pluck('id')->all();
+        $missingIds = array_values(array_diff($policyIds, $existingIds));
+
+        if (!empty($missingIds)) {
+            return response()->json([
```
Comment (tarrow) [inline app/Http/Controllers/PolicyAcceptanceController.php:24]: I looked and thought should we `return response()->json(..)` or throw here? I checked what is the standard and we regularly do this so let's keep this pattern.

File: tests/Routes/PolicyAcceptanceControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    public function testPolicyIdsNotAnArrayReturns422(): void {
+        $user = User::factory()->create();
+
+        $this->actingAs($user, 'api')
+            ->json('PUT', $this->route, ['policy_ids' => 'not-an-array'])
+            ->assertStatus(422)
+            ->assertJsonStructure(['errors' => ['policy_ids']]);
+    }
```
Comment (tarrow) [inline tests/Routes/PolicyAcceptanceControllerTest.php:141]: I think these tests are very thorough but I did also realise you could test for the "non-integer policy id" situation that you mentioned in the PR description.

Not a blocker to merging if you don't want to add it though.

File: app/Http/Controllers/PolicyAcceptanceController.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            ], 400);
+        }
+
+        $userId = $request->user()->id;
+
+        foreach ($policyIds as $policyId) {
+            // Ignore if the user has already accepted this policy
+            PolicyAcceptance::firstOrCreate(
+                ['user_id' => $userId, 'policy_id' => $policyId],
+                ['accepted_at' => now()],
```
Comment (tarrow) [inline app/Http/Controllers/PolicyAcceptanceController.php:37]: Does this do what we want? By reading it (but not executing and trying it out) I'm worried this updates the `accepted_at` when it should remain at the old time. Perhaps we could confirm this by updating the test (comment added below)

File: tests/Routes/PolicyAcceptanceControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            'policy_id' => $policy->id,
+            'accepted_at' => now(),
+        ]);
+
+        $this->actingAs($user, 'api')
+            ->json('PUT', $this->route, ['policy_ids' => [$policy->id]])
+            ->assertStatus(200)
+            ->assertJson(['success' => true]);
+
+        $this->assertSame(1, PolicyAcceptance::where([
```
Comment (tarrow) [inline tests/Routes/PolicyAcceptanceControllerTest.php:77]: Could we test that the `accepted_at` value isn't changed here? (see above)

**Review by tarrow — CHANGES_REQUESTED** [review]
Only blocker from my side is around confirming we don't update `accepted_at` time on each request.

File: tests/Routes/PolicyAcceptanceControllerTest.php (hunk 1 above)
Comment (dati18) [reply tests/Routes/PolicyAcceptanceControllerTest.php:37]: The second argument `api` is telling Laravel's auth manager that "for this test duration, set this user as an _authenticated user_ for the `api` guard". When you just call `$this->actingAs($user)`, Laravel logs that user into default guard (I haven't checked what is the "default", usually cookie auth?).
We have been using this pattern for testing purposes, basically bypassing all the hassle of authenticating the normal way

File: tests/Routes/PolicyAcceptanceControllerTest.php (hunk 6 above)
Comment (dati18) [reply tests/Routes/PolicyAcceptanceControllerTest.php:77]: As I understand, `firstOrCreate()` function doesn't do row update if row is already there. Check [this](https://laravel.com/docs/13.x/eloquent#:~:text=or%20Creating%20Models-,The,merging%20the%20first%20array%20argument%20with%20the%20optional%20second%20array%20argument.,-The)

But I kind of understood your standpoint, so I created a test for edge case like this.

File: tests/Routes/PolicyAcceptanceControllerTest.php (hunk 1 above)
Comment (tarrow) [reply tests/Routes/PolicyAcceptanceControllerTest.php:37]: Yes, I was wondering if we could either use the default guard or make `api` the default guard. You're right this is the same pattern we use in other places and is a good pattern to follow I think

File: tests/Routes/PolicyAcceptanceControllerTest.php (hunk 6 above)
Comment (tarrow) [reply tests/Routes/PolicyAcceptanceControllerTest.php:77]: Thanks, looks like ` updateOrCreate()` would have been the function that would get and potentially update it.

File: routes/api.php [hunk 7]
```diff
@@ -28,6 +28,9 @@
         $router->get('auth/login', ['uses' => 'Auth\LoginController@getLogin']);
         $router->delete('auth/login', ['uses' => 'Auth\LoginController@deleteLogin']);
 
+        // policy acceptances
+        $router->put('policy_acceptances', ['uses' => 'PolicyAcceptanceController@store']);
```
Comment (tarrow) [inline routes/api.php]: I realise I missed one small thing in my previous review (sorry). The ticket asks for this api to be under `/v1/` rather than in the root.

This is a pattern we are not yet following everywhere but something we would like to do.
```suggestion
        $router->put('v1/policy_acceptances', ['uses' => 'PolicyAcceptanceController@store']);
```

Comment (dati18) [reply routes/api.php]: ah a blunder... fixed

**Review by tarrow — APPROVED** [review]
Thanks for the final change. I'm now happy with this.
